zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/JsonReaders.java

```diff
@@ -101,5 +99,24 @@ static void visitNextOrSkip(JsonParser parser, String name, Set<String> result)
     }
   }
 
-  JsonReaders() {}
+  static boolean checkStartObject(JsonParser parser, boolean shouldThrow) throws IOException {
+    try {
+      JsonToken currentToken = parser.currentToken();
+      // The parser may not be at a token, yet. If that's the case advance.
+      if (currentToken == null) currentToken = parser.nextToken();
+
+      // If we are still not at the expected token, we could be an another or an empty body.
+      if (currentToken == JsonToken.START_OBJECT) return true;
+      if (shouldThrow) {
+        throw new IllegalArgumentException("Expected start object, was " + currentToken);
+      }
+      return false;
+    } catch (JsonParseException e) { // likely not json
```

Comment:
catch `Throwable`? When methods shouldn't throw, like where `shouldThrow` is `false` here, it's easy to end up with a hang because an exception type changed.

Reply:
yep!